### PR TITLE
v2v: rhv plugin - find suitable host

### PR DESCRIPTION
**Pull request by danielerez**
For direct upload, a suitable host must be in status 'Up'
and belong to the same datacenter as the created disk.
Added these criteria to the host search query.

### CODE REVIEW

**Comment by danielerez** [comment]
@rwmjones 
@nirs

File: v2v/rhv-upload-plugin.py [hunk 1]
```diff
@@ -67,11 +67,23 @@ def find_host(connection):
...
-    debug("hw_id = %r" % vdsm_id)
+    system_service = connection.system_service()
+    storage_name = params['output_storage']
+    data_centers = system_service.data_centers_service().list(
+        search='storage=%s' % storage_name,
+        case_sensitive=False,
+    )
+    if len(data_centers) == 0:
+        # The storage domain is not attached to a datacenter
+        # (shouldn't happen, would fail on disk creation).
```
Comment (nirs) [inline v2v/rhv-upload-plugin.py]: I think we need a debug message here, similar to the one when vdsm_id is missing.

File: v2v/rhv-upload-plugin.py [hunk 2]
```diff
@@ -67,11 +67,23 @@ def find_host(connection):
         debug("cannot read /etc/vdsm/vdsm.id, using any host: %s" % e)
         return None
 
-    debug("hw_id = %r" % vdsm_id)
```
Comment (nirs) [inline v2v/rhv-upload-plugin.py]: I would leave this as is.

File: v2v/rhv-upload-plugin.py [hunk 3]
```diff
@@ -67,11 +67,23 @@ def find_host(connection):
...
+        return None
+
+    datacenter = data_centers[0]
+    debug("hw_id = %r, datacenter = %s" % (vdsm_id, datacenter.name))
 
-    hosts_service = connection.system_service().hosts_service()
+    hosts_service = system_service.hosts_service()
     hosts = hosts_service.list(
-        search="hw_id=%s" % vdsm_id,
+        search="hw_id=%s and datacenter=%s and status=Up" % (vdsm_id, datacenter.name),
```
Comment (nirs) [inline v2v/rhv-upload-plugin.py]: We can replace the debug message bellow with something like:

   host with hw_id=%s cannot upload to storage domain %s, using any host

**Comment by libguestfs** [comment]
Continuing discussion on the mailing list:
https://www.redhat.com/archives/libguestfs/2018-July/msg00008.html

**Pull request closed without merging**